message.module

```diff
@@ -64,37 +66,43 @@ function message_entity_delete(EntityInterface $entity) {
     }
 
     if ($field->getCardinality() == 1) {
-      $deletable_mids += array_keys($results);
+      $deletable_mids += $results;
```

Comment:
Correct. If the docs for entity query `execute()` are to be believed:
```
The values of the array are always entity ids. The keys will be revision ids if the entity supports revision and entity ids if not.
```
Messages aren't revisionable, so each key should be the same as its value, so if there are collisions, it doesn't matter that one is being ignored. The trouble with using `array_keys()` is that what's being returned is a zero-indexed array, and so the next single valued field _will_ have things getting ignored that should not be.

But maybe it would be better to make it a little more safe/explicit and switch to `array_merge()` of the array's values.